randovania/game_description/db/hint_node.py

```diff
@@ -54,3 +52,92 @@ def is_collected(self, context: NodeContext) -> bool:
 
     def resource_gain_on_collect(self, context: NodeContext) -> ResourceGain:
         yield self.resource(context), 1
+
+    def data_editor_message(self, region_list: RegionList, db: ResourceDatabase) -> str:
+        raise NotImplementedError
+
+
+@dataclasses.dataclass(frozen=True, slots=True)
+class GenericHintNode(HintNode):
+    @classmethod
+    def hint_kind(cls) -> HintNodeKind:
+        return HintNodeKind.GENERIC
+
+    def data_editor_message(self, region_list: RegionList, db: ResourceDatabase) -> str:
+        message = "Generic Hint"
+        if (requirement := self.requirement_name) != "Trivial":
+            message += f"\n<br />Requirement: {requirement}"
+        return message
+
+
+@dataclasses.dataclass(frozen=True, slots=True)
+class SpecificLocationHintNode(HintNode):
+    @classmethod
+    def hint_kind(cls) -> HintNodeKind:
+        return HintNodeKind.SPECIFIC_LOCATION
+
+    @property
+    def target_index(self) -> PickupIndex:
+        """Which PickupIndex this hint points to."""
+        return PickupIndex(self._target)
+
+    def data_editor_message(self, region_list: RegionList, db: ResourceDatabase) -> str:
+        message = "Specific Location Hint"
+        target = region_list.node_from_pickup_index(self.target_index)
+        fmt_message = '\n<br />Target: <a href="node://{}">{}</a>'
```

Comment:
I don't like having effectively GUI code right up into the core classes

Reply:
could do it as a series of `isinstance()` checks where it's actually used, if you prefer